src/main/java/net/glowstone/net/handler/play/player/PlayerAbilitiesHandler.java

```diff
@@ -15,7 +17,13 @@ public void handle(GlowSession session, PlayerAbilitiesMessage message) {
 
         GlowPlayer player = session.getPlayer();
         boolean flying = (message.getFlags() & 0x02) != 0;
-
-        player.setFlying(player.getAllowFlight() && flying);
+        boolean isFlying = player.isFlying();
+        if (isFlying != flying) {
+            if (!flying || player.getAllowFlight()) {
```

Comment:
Add a comment describing the logic for this if-statement.

e.g. `// fire the event if either the player disabled flying, or enabled it and is allowed to fly`